pkg/install/generate.go

```diff
@@ -575,8 +585,9 @@ func GenerateInstallerJob(provision *hivev1.ClusterProvision) (*batchv1.Job, err
 			Labels:    labels,
 		},
 		Spec: batchv1.JobSpec{
-			BackoffLimit: pointer.Int32Ptr(0),
-			Completions:  pointer.Int32Ptr(1),
+			BackoffLimit:          pointer.Int32Ptr(0),
+			Completions:           pointer.Int32Ptr(1),
+			ActiveDeadlineSeconds: pointer.Int64Ptr(int64(provisionJobDeadline.Seconds())),
```

Comment:
When this procs the the ClusterProvision shows the following:

```yaml
  - lastProbeTime: "2021-04-08T17:27:33Z"
    lastTransitionTime: "2021-04-08T17:27:33Z"
    message: Cluster install failed but installer log was not captured
    reason: UnknownError
    status: "True"
    type: ClusterProvisionFailed
```

Would it be possible to get some detail in there to clarify it hit the deadline. The linked job does have this detail available and we could copy it back. 

```yaml
status:
  conditions:
  - lastProbeTime: "2021-04-08T17:27:33Z"
    lastTransitionTime: "2021-04-08T17:27:33Z"
    message: Job was active longer than specified deadline
    reason: DeadlineExceeded
    status: "True"
    type: Failed
```

Reply:
the provision also updates the reason for deadline exceeded.

```
$ oc get clusterprovision test-cluster-0-9gj4q -ojson | jq .status.conditions
[
  {
    "lastProbeTime": "2021-04-13T21:40:45Z",
    "lastTransitionTime": "2021-04-13T21:40:45Z",
    "message": "Install job has been created",
    "reason": "JobCreated",
    "status": "True",
    "type": "ClusterProvisionJobCreated"
  },
  {
    "lastProbeTime": "2021-04-13T21:41:11Z",
    "lastTransitionTime": "2021-04-13T21:41:11Z",
    "message": "Install job has completed its initialization. Provisioning started.",
    "reason": "InitializationComplete",
    "status": "True",
    "type": "ClusterProvisionInitialized"
  },
  {
    "lastProbeTime": "2021-04-13T21:43:45Z",
    "lastTransitionTime": "2021-04-13T21:43:45Z",
    "message": "Install job failed due to deadline being exceeded for the attempt",
    "reason": "AttemptDeadlineExceeded",
    "status": "True",
    "type": "ClusterProvisionFailed"
  }
]

$ oc get cd test-cluster -ojson | jq .status.conditions
[
  {
    "lastProbeTime": "2021-04-13T21:40:37Z",
    "lastTransitionTime": "2021-04-13T21:40:37Z",
    "message": "ClusterImageSet test-cluster-imageset is available",
    "reason": "ClusterImageSetFound",
    "status": "False",
    "type": "ClusterImageSetNotFound"
  },
  {
    "lastProbeTime": "2021-04-13T21:43:45Z",
    "lastTransitionTime": "2021-04-13T21:43:45Z",
    "message": "Provision test-cluster-0-9gj4q failed. Next provision at 2021-04-13T21:44:45Z.",
    "reason": "AttemptDeadlineExceeded",
    "status": "True",
    "type": "ProvisionFailed"
  }
]

```